source/renderer/app/api/api.js

```diff
@@ -836,6 +837,22 @@ export default class AdaApi {
       return _createTransactionFromServerData(response);
     } catch (error) {
       logger.error('AdaApi::createTransaction error', { error });
+      const adaToProceedRegex = /(?:.*I need approximately (.*) ada to proceed.*)/;
```

Comment:
I think we can use a simpler more efficient regex here

This one if the ada amount is whole:
`/I need approximately([\s\d]+)ada to proceed/`

This one if the ada amount is float (with dot or comma for separator)
`/I need approximately([\s\d\.,]+)ada to proceed/`

I did a test and these take around 40 steps, against +400 steps of the original.

Reply:
I might update this PR as well:
https://github.com/input-output-hk/daedalus/pull/2830/files#diff-2c52a7019186cbaf59d4df33d89de827d3cd45fb46142b483e769d186dfa3af9R30